t_sites/test_tempesta_tech.py

```diff
@@ -0,0 +1,482 @@
+        }
+        response = self.send_request(client)
+        self.assertEqual(response.status, 201)
+        self.assertFalse(self.check_cached_headers(response.headers))
+        post_id = response.headers["location"].split("/")[-1]
+        return post_id
+
+    def get_comments_feed(self):
+        return self.get_page_content("/comments/feed/")
+
+    def post_comment(self, post_id, text="Test", anonymous=True):
+        data = (
+            f"comment={text}"
+            "&author=anonymous"
+            "&email=guest%40example.com"
+            "&url="
+            "&submit=Post+Comment"
+            f"&comment_post_ID={post_id}"
+            "&comment_parent=0"
+        )
+        response = self.post_form(uri="/wp-comments-post.php", data=data, anonymous=anonymous)
+        self.assertEqual(response.status, 302, response)
+        self.assertFalse(self.check_cached_headers(response.headers))
+
+    def post_form(self, uri, data, anonymous=True):
+        client = self.get_client("post_form" if anonymous else "post_admin_form")
+        client.load_cookies = not anonymous
+        client.set_uri(uri)
+        client.data = data
+        response = self.send_request(client)
+        self.assertFalse(self.check_cached_headers(response.headers))
+        return response
+
+    def get_post(self, post_title):
+        client = self.get_client("get")
+        client.set_uri(f"/uncategorized/{post_title}/")
+        response = self.send_request(client)
+        self.assertEqual(response.status, 200)
+        return response
+
+    def test_blog_post_flow(self):
+        """
+        - add a new post page;
+        - add a new comment to this page;
+        - update cache;
+        - cache the comment is present on this page;
+        """
+        post_title = f"{time.time()}_post_title"
+        user_comment = f"{time.time()}_user_comment"
+        self.start_all_services(client=False)
+
+        # Login
+        self.login()
+
+        # Obtain nonce
+        nonce = self.get_nonce()
+
+        # Publish new blog post
+        post_id = self.post_blog_post(title=post_title, nonce=nonce)
+
+        # Get new blog post content
+        content = self.get_page_content(f"/uncategorized/{post_title}/")
+        self.assertIn(post_title, content, f"The {post_title} is not preset on blog post.")
+
+        # No comments yet
+        self.assertNotIn(user_comment, content)
+        feed = self.get_comments_feed()
+        self.assertNotIn(user_comment, feed)
+
+        # Post comment
+        self.post_comment(post_id, anonymous=False, text=user_comment)
+        response = self.get_post(post_title)
+        # Check comment is not present because response from cache
+        self.assertNotIn(user_comment, response.stdout)
+        self.assertNotIn(user_comment, self.get_comments_feed())
+
+        # Purge cache
+        self.purge_cache(f"/uncategorized/{post_title}/", fetch=True)
+        self.purge_cache("/comments/feed/", fetch=True)
```

Comment:
It seems this is wrong: a browser sends `Cache-Control: no-cache`, so Tempesta must revalidate the blog post and reply with a fresh copy of the blog post page (see https://github.com/tempesta-tech/tempesta-tech.com/issues/64#issuecomment-2141852434) WITOHUT the cache purge. Could you please check this with the last master with https://github.com/tempesta-tech/tempesta/issues/1995 fixed?

@EvgeniiMekhanik @const-t FYI: if we still have the behavior on the current master, then we have a caching bug. Or I misunderstand `Cache-Control: no-cache`...

Reply:
It seems we need to preserve old copies by RFC, so please comment which RFC requires this and check Nginx and Varnish as the strongest webcache implementation

Reply:
For comments by not authorized users:

I create comment for `/blog/building-your-own-wordpress-staging-with-tempesta-fw/` and wordpress redirects me to urn `/blog/building-your-own-wordpress-staging-with-tempesta-fw/?unapproved=11&moderation-hash=cdfce7ef1667e12e7368d14d27284a2e#comment-11`. The requests after redirect contain `Cache-control: no-cache/max-age=0` header for different browsers (Chrome/Opera/FireFox). 

But the response will be not from cache when I make request to `/blog/building-your-own-wordpress-staging-with-tempesta-fw/` again because uri with other query params is a new uri. We have [tests](https://github.com/tempesta-tech/tempesta-test/blob/df8a0e7ec6f4a4cec96666db9f953490a710c564/cache/test_cache.py#L352) for that.

Then a moderator approves the comment and I request to urn `/blog/building-your-own-wordpress-staging-with-tempesta-fw/` again. As I understand it, the response will be from cache. But maybe the cache will be updated after the approve of comment. I would like check it, but I need access.

Let's look at another case:
Website does not redirect me another urn (example for github). It just add comment to PR/issue via JS. And the responses for PR/issue have `Cache-Control: no-cache/max-age=0` header. So these pages will never be cached.

Reply:
I read RFC 9111 and I see there is only one sense to save several responses with same method for one URI - Vary header. Since we don't support Vary header now we can delete previously saved responses with same method.  
About Cache-Control: no-cache currently we request a new copy of upstream, because we have special issue https://github.com/tempesta-tech/tempesta/issues/518 for conditional requests, which is currently not implemented

Reply:
I'm still working on the issue, but today I focused only on commenting from under an authenticated user (see my comment proposal above). The manual steps and results are:

1. populate the Tempesta cache with a blog post page
2. start a private mode firefox, open web developer tools and make sure `Disable Cache` isn't enabled
3. login to `wp-admin`, open a new tab and go to the blog post - I see the WP admin panel on the top of the blog post page
4. post a comment and get the redirected page
5. here I see the page **without** the admin panel above and **without** my comment

Step 5 is different in both the points from what I see if I work with the website container without Tempesta.

One strange thing is that on step 1 if I use Firefox private sessions, then I always see cache misses on Tempesta accounting and only `curl` fetches the page from the cache.

Reply:
The problem scenario from the above isn't reproduced if `Disable Cache` is switched on in a browser. This option is not only about browser caching, but a browser (tested with Firefox) also adds `Cache-Control: no-cache` to all requests.

Reply:
I tried to use `cache_disable`  but it has not effect, so I leave the test unchanged.